src/main/java/org/opensearch/knn/indices/ModelDao.java

```diff
@@ -330,13 +330,7 @@ private void putInternal(Model model, ActionListener<IndexResponse> listener, Do
                 );
             }, listener::onFailure);
 
-            // After the model is indexed, update metadata only if the model is in CREATED state
-            ActionListener<IndexResponse> onIndexListener;
-            if (ModelState.CREATED.equals(model.getModelMetadata().getState())) {
-                onIndexListener = getUpdateModelMetadataListener(model.getModelMetadata(), onMetaListener);
-            } else {
-                onIndexListener = onMetaListener;
-            }
+            ActionListener<IndexResponse> onIndexListener = getUpdateModelMetadataListener(model.getModelMetadata(), onMetaListener);
```

Comment:
Want to discuss a little bit here some thoughts:

So before, the logic was that the model index is the only source of truth until the model is actually created. Then the model-metadata is a source of truth as well.

Now, we are saying that the model metadata will always lag behind the model system index (i.e. on any call to change model state, we will first persist in system index and then update in cluster state). This is going to open up to the possibility that model index and cluster state fall out of sync on failure. We may need to think about how we handle different scenarios. 

I think a model's state can be updated in one of the following ways:
1. Training process (this will be in sync - single thread, synchronous process)
2. Model deleted (we block the model delete if the model is not in the created or error state - we should confirm that we check this from cluster metadata before blocking)
3. Node drop (if a node drops, the offending node will know not to change anything in the state correct?)

Are there other cases you can think of that might be of concern?

Reply:
@jmazanec15 The goal of this was to treat the cluster metadata as the main source of truth in case of node drops I believe

Reply:
I wrote a temporary integration test trying to create a race condition between a model finishing training and entering the created state in ModelDao and then deleting the model immediately.  All behavior was as expected and the model was deleted from the cluster metadata as well.

Reply:
@ryanbogan can you push to another branch and link to the test? I want to take a quick look.